src/main/java/com/epam/aidial/core/config/FileConfigStore.java

```diff
@@ -98,6 +103,16 @@ private void load(boolean fail) {
         }
     }
 
+    private void associateDeploymentWithApiKey(Config config, Deployment deployment) {
+        if (deployment.getApiKey() == null) {
+            deployment.setApiKey(generateKey());
+        }
+        Key key = new Key();
+        key.setKey(deployment.getApiKey());
+        key.setProject(deployment.getName());
+        config.getKeys().put(deployment.getApiKey(), key);
```

Comment:
Please assert that keys are never overwritten.

Reply:
fixed

Reply:
It had better generate a new one.

Reply:
fixed

Reply:
fixed